packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -427,39 +427,78 @@ class ProductVariantInventoryService extends TransactionBaseService {
       .listReservationItems(
         {
           line_item_id: lineItemId,
+          location_id: locationId,
         },
         {
           order: { created_at: "DESC" },
         }
       )
 
-    if (reservationCount) {
-      let reservation = reservations[0]
+    if (quantity > 0) {
```

Comment:
question (non-blocking): are we making positive adjustments at this point? If not, what are you envisioning that the use case for this will be?

Reply:
@pKorsholm still curious to hear your thoughts

Reply:
ah yeah, no we are not using this at any point at the moment, we could just remove this clause and throw instead chalking it up to undefined behavior. We just have to handle it somehow since the remaining code assumes the difference is negative. 
 wdyt?

Reply:
Would maybe be safer to just throw with "You can only reduce reservation quantities using adjustReservationsQuantityByLineItem. If you wish to reserve more use update or create."

But no strong feelings